src/ja/user/cli.py

```diff
@@ -1,19 +1,244 @@
         @return: The ServerCommand created from the combined input.
         """
+
+        with open(self._config_path, "r") as stream:
+            data_loaded = yaml.safe_load(stream)
+
+        # Creating a UserConfig instance from the user config file.
+        verbosity: int = 1 if "verbosity" not in data_loaded else data_loaded["verbosity"]
+        hostname: str = None if "hostname" not in data_loaded else data_loaded["hostname"]
+        username: str = None if "username" not in data_loaded else data_loaded["username"]
+        password: str = None if "password" not in data_loaded else data_loaded["password"]
+        key_path: str = None if "key_path" not in data_loaded else data_loaded["key_path"]
+        passphrase: str = None if "passphrase" not in data_loaded else data_loaded["passphrase"]
+        email: str = None if "email" not in data_loaded else data_loaded["email"]
+
+        parser: ArgumentParser = ArgumentParser("JobAdder: add, cancel or a query a job.")
+        # shared options
+        parser.add_argument("--verbosity", "-v", type=int, choices=[0, 1, 2], default=verbosity)
+        parser.add_argument("--hostname", "--host", type=str,
+                            required=hostname is None, default=hostname,
+                            help="The name of the host to connect to.")
+        parser.add_argument("--username", "--user", type=str,
+                            required=username is None, default=username,
+                            help="The name of the user to use for login.")
+        parser.add_argument("--password", "--pass", type=str, default=password, help="The password to use for login.")
+        parser.add_argument("--key-path", "--kp", type=str, default=key_path, help="The key pair to use for login.")
+        parser.add_argument("--passphrase", "--pp", type=str, default=passphrase,
+                            help="The passphrase to use for decrypting private keys.")
+
+        subparsers = parser.add_subparsers(dest="command")
+        # parser for add command.
+        parser_add = subparsers.add_parser("add")
+        parser_add.add_argument("--config", "-c", help="Config file path for adding a job.")
+        parser_add.add_argument("--label", "-l", help="Label for the job.")
+        parser_add.add_argument("--source", "-s", "--path", help="Path of the source file containing the job.")
+        parser_add.add_argument("--priority", "-p", choices="low medium high urgent 0 1 2 3".split())
+        parser_add.add_argument("--threads", "-t", type=int, help="Assign a number of cpu threads for your job.")
+        parser_add.add_argument("--memory", "-m", type=int, help="Assign a specific amount of memory to your job")
+        parser_add.add_argument("--preemptible", "--pe", action="store_true")
+        parser_add.add_argument("--blocking", "--bl", action="store_true")
+        parser_add.add_argument("--mount", nargs="+", type=str)
```

Comment:
I think it would be nice to add `action='append'`, this way the user can run `--mount A B --mount C D` which makes it more clear (also, this way you can just specify `nargs='2'` to make sure you always get pairs)